src/support/z_datamodel.erl

```diff
@@ -50,12 +50,14 @@ manage(Module, Datamodel, Context) ->
 -spec manage(atom(), #datamodel{}, datamodel_options(), #context{}) -> ok.
 manage(Module, Datamodel, Options, Context) ->
     AdminContext = z_acl:sudo(Context),
-    [manage_category(Module, Cat, Options, AdminContext) || Cat <- Datamodel#datamodel.categories],
-    [manage_predicate(Module, Pred, Options, AdminContext) || Pred <- Datamodel#datamodel.predicates],
-    [manage_resource(Module, R, Options, AdminContext) || R <- Datamodel#datamodel.resources],
-    [manage_medium(Module, Medium, Options, AdminContext) || Medium <- Datamodel#datamodel.media],
-    [manage_edge(Module, Edge, Options, AdminContext) || Edge <- Datamodel#datamodel.edges],
-    [manage_data(Module, Data, AdminContext) || Data <- Datamodel#datamodel.data],
+    jobs:run(manage_module_jobs, fun() ->
+        [manage_category(Module, Cat, Options, AdminContext) || Cat <- Datamodel#datamodel.categories],
+        [manage_predicate(Module, Pred, Options, AdminContext) || Pred <- Datamodel#datamodel.predicates],
+        [manage_resource(Module, R, Options, AdminContext) || R <- Datamodel#datamodel.resources],
+        [manage_medium(Module, Medium, Options, AdminContext) || Medium <- Datamodel#datamodel.media],
+        [manage_edge(Module, Edge, Options, AdminContext) || Edge <- Datamodel#datamodel.edges],
+        [manage_data(Module, Data, AdminContext) || Data <- Datamodel#datamodel.data]
+    end),
```

Comment:
I saw some tests running into a situation where multiple *category:renumber* jobs were installed in parallel. Apparently we can have multiple modules being enabled at the same time. This serializes all changes, which prevents all kinds of race conditions.

To be fair, I think we should add a serialization on the hierarchy changes as well.